contracts/src/AttestationRegistry.sol

```diff
@@ -182,29 +184,34 @@ contract AttestationRegistry is OwnableUpgradeable {
    * @notice Replaces attestations for given identifiers and replaces them with new attestations
    * @param attestationIds the list of IDs of the attestations to replace
    * @param attestationPayloads the list of attestation payloads to create the new attestations and register them
-   * @param attester the account address issuing the attestation
+   * @param replacer the account address replacing the attestation
    */
   function bulkReplace(
     bytes32[] calldata attestationIds,
     AttestationPayload[] calldata attestationPayloads,
-    address attester
+    address replacer
   ) public {
     if (attestationIds.length != attestationPayloads.length) revert ArrayLengthMismatch();
     for (uint256 i = 0; i < attestationIds.length; i = uncheckedInc256(i)) {
-      replace(attestationIds[i], attestationPayloads[i], attester);
+      replace(attestationIds[i], attestationPayloads[i], replacer);
     }
   }
 
   /**
    * @notice Revokes an attestation for a given identifier
    * @param attestationId the ID of the attestation to revoke
+   * @param revoker the account address revoking the attestation
    */
-  function revoke(bytes32 attestationId) public {
+  function revoke(bytes32 attestationId, address revoker) public {
```

Comment:
If the `revoker` argument is being specified by the contract calling this method, then surely it can be spoofed? Am I wrong?

Reply:
Actually, the Attestation registry will always be called from the portal that has implemented the abstract portal. The abstract portal has revoke and replace methods that cannot be overridden. The abstract portal has getAttester method which returns msg sender.

As it is recommended not to use tx origin, the best way is to get the attester from the Abstract portal.